### PR TITLE
Unity Task environment setter

### PR DESCRIPTION
**Pull request by Azurelol**
## Description
Add a property for setting the environment for an Unity Task.

## Changes
* ![UPDATE] UnityTask
* ![UPDATE] ArgumentsSpec

[NEW]:https://atlas-resources.wooga.com/icons/icon_new.svg "New"
[ADD]:http://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:http://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:http://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:http://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:http://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:http://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:http://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:http://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:http://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:http://resources.atlas.wooga.com/icons/icon_webGL.svg "Web:GL"

### CODE REVIEW

File: src/integrationTest/groovy/wooga/gradle/unity/UnityIntegrationSpec.groovy [hunk 1]
```diff
@@ -139,12 +139,14 @@ abstract class UnityIntegrationSpec extends IntegrationSpec {
                 @echo off
                 echo ${mockUnityStartupMessage}
                 echo %*
+                echo env
```
Comment (Larusso) [inline src/integrationTest/groovy/wooga/gradle/unity/UnityIntegrationSpec.groovy:142]: is `echo env` the same on windows?

File: src/main/groovy/wooga/gradle/unity/UnityTask.groovy [hunk 2]
```diff
@@ -63,14 +64,16 @@ abstract class UnityTask extends DefaultTask
 
                 def unityPath = unityPath.get().asFile.absolutePath
                 def unityArgs = getAllArguments()
-
-                OutputStream outputStream = getOutputStream()
+                def unityEnvironment = environment.get()
+                def outputStream = getOutputStream()
+                environment.get().sort()
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/UnityTask.groovy:69]: what is the call to sort for? Is it in place?

Comment (Azurelol) [reply src/main/groovy/wooga/gradle/unity/UnityTask.groovy:69]: This wasn't meant to be committed (was checking functions); my apologies

File: src/integrationTest/groovy/wooga/gradle/unity/UnityIntegrationSpec.groovy (hunk 1 above)
Comment (Azurelol) [reply src/integrationTest/groovy/wooga/gradle/unity/UnityIntegrationSpec.groovy:142]: No it's not;; I will look for the right invocation.

Comment (Larusso) [reply src/integrationTest/groovy/wooga/gradle/unity/UnityIntegrationSpec.groovy:142]: It's `set` as far as I can remember.

Comment (Azurelol) [reply src/integrationTest/groovy/wooga/gradle/unity/UnityIntegrationSpec.groovy]: Aye that's correct.

File: src/main/groovy/wooga/gradle/unity/UnityTask.groovy [hunk 3]
```diff
@@ -63,14 +64,15 @@ abstract class UnityTask extends DefaultTask
 
                 def unityPath = unityPath.get().asFile.absolutePath
                 def unityArgs = getAllArguments()
-
-                OutputStream outputStream = getOutputStream()
+                def unityEnvironment = environment.get()
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/UnityTask.groovy:67]: As I mentioned. The `MapProperty` class implements the `Map` interface. I don't think it is needed to fetch it explicitly. Or does it throw fits?

Comment (Azurelol) [reply src/main/groovy/wooga/gradle/unity/UnityTask.groovy:67]: The exec call's doesn't is of Map type (no property) so I had to pass the unwrapped value.

Comment (Azurelol) [reply src/main/groovy/wooga/gradle/unity/UnityTask.groovy:67]: image

Comment (Larusso) [reply src/main/groovy/wooga/gradle/unity/UnityTask.groovy:67]: ok then.

**Review by Larusso — APPROVED** [review]

**Pull request merged**